### DISCUSSION TITLE
主菜单的二级分类，在文章设置中不显示层级关系。删除后依然会显示

### DISCUSSION THREAD

**Discussion by mengtaixiao** [halo-dev/halo#6460] [Q&A]
### 系统信息

- 外部访问地址: 
- 启动时间: 2024-08-12 03:29
- 版本: 2.18.0
- 构建时间: 2024-08-01 10:17
- Git Commit: 58fe872
- Java: IBM Semeru Runtime Open Edition / 21.0.3+9-LTS
- 数据库: H2 / 2.2.224
- 操作系统: Linux / 6.1.0-20-amd64
- 已激活主题: [Earth 1.7.1](https://github.com/halo-dev/theme-earth.git)
- 已启动插件:
  - [Sitemap 1.1.2](https://github.com/halo-dev/plugin-sitemap)
  - [搜索组件 1.5.0](https://github.com/halo-dev/plugin-search-widget)
  - [RSS 1.3.0](https://github.com/halo-dev/plugin-feed)
  - [评论组件 2.4.0](https://github.com/halo-dev/plugin-comment-widget)
  - [应用市场 1.2.0](https://www.halo.run/store/apps/app-VYJbF)

### 使用的哪种方式运行？

Docker

### 发生了什么？

我在Halo后台--菜单--主菜单，创建了一级分类[游戏]，在一级分类[游戏]下创建了二级分类[手机游戏]
![image](https://github.com/user-attachments/assets/14301164-3266-4c1d-a133-434b332eb345)
但是当我回到文章设置，设置文章的分类时，这两个分类，却没有层级关系
![image](https://github.com/user-attachments/assets/0c07dc57-db4e-4ca5-aa91-cfe6d1223f8c)

我在以前版本的Halo中是创建出来过有层级关系的分类的，但现在却不行了。拖动改变层级关系，也不会在文章设置中体现出来
![image](https://github.com/user-attachments/assets/77601a68-36b0-4137-ac3d-55a65bf3129c)

![image](https://github.com/user-attachments/assets/d139523f-306b-426e-8c03-7595a88848d5)

然后我尝试创建并删除了一些多级分类，然后立即删除这些分类，但在文章设置中，依然能看到这些分类，并且不知道怎么删除
![image](https://github.com/user-attachments/assets/1cab35d9-6698-4975-b8df-f5eb83ad2f2d)
![image](https://github.com/user-attachments/assets/6302aa6a-db56-4847-ab43-2586acb49033)
像这样，在文章设置中多了很多测试用的分类，明明主菜单中已经删掉了的，但这里还是会显示出来，也找不到哪里再次删除。
尝试过强制刷新，新开无痕模式，重启Halo，但问题依旧

我以为是我旧版本升级的问题，但我在全新安装的2.18版本中可以复现该问题

**Comment by ruibaby** [comment]
菜单是菜单，分类是分类，文章设置这里是分类而不是菜单。

/kind support
/triage unresolved

**Comment by mengtaixiao** [comment]
> 菜单是菜单，分类是分类，文章设置这里是分类而不是菜单。
> 
> /kind support /triage unresolved

那么我该如何操作才能实现，在文章设置中，添加出这样带层级的 分类呢
以及，如何删除分类，我在菜单那边是已经删掉了的
![image](https://github.com/user-attachments/assets/3601d875-0f7a-4b93-a2cd-03e7f235d542)

**Comment by guqing** [comment]
> 那么我该如何操作才能实现，在文章设置中，添加出这样带层级的 分类呢 以及，如何删除分类，我在菜单那边是已经删掉了的 !

拖拽缩进即可

<img width="554" alt="image" src="https://github.com/user-attachments/assets/28171882-cd33-471d-8d6a-41557fec06ec">

**Comment by mengtaixiao** [comment]
> > 那么我该如何操作才能实现，在文章设置中，添加出这样带层级的 分类呢 以及，如何删除分类，我在菜单那边是已经删掉了的 !
> 
> 拖拽缩进即可
> 
> <img alt="image" width="554" src="https://private-user-images.githubusercontent.com/38999863/356926979-28171882-cd33-471d-8d6a-41557fec06ec.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM0MzI0NjksIm5iZiI6MTcyMzQzMjE2OSwicGF0aCI6Ii8zODk5OTg2My8zNTY5MjY5NzktMjgxNzE4ODItY2QzMy00NzFkLThkNmEtNDE1NTdmZWMwNmVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODEyVDAzMDkyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJlODg1OGQ4ZjliY2UxNzNkYWM2ZDI3NjU2N2FjMmNjMzViOTE3MjI0M2M2OGFmNDZjODI2NTBiOGNmMjljZWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.eIIvZAic3c9XK2REosYJcjG88cdDjLMmFlttNcRbxLQ">

不行的啊
![image](https://github.com/user-attachments/assets/65a4459c-2e12-40f5-a953-3eb10511102c)

![image](https://github.com/user-attachments/assets/5775e303-e451-4397-a68a-76e5497f8420)

**Comment by ruibaby** [comment]
看文档：https://docs.halo.run/user-guide/posts#%E6%96%87%E7%AB%A0%E5%88%86%E7%B1%BB%E7%AE%A1%E7%90%86

**Comment by guqing** [comment]
> > > 那么我该如何操作才能实现，在文章设置中，添加出这样带层级的 分类呢 以及，如何删除分类，我在菜单那边是已经删掉了的 !
> > 
> > 
> > 拖拽缩进即可
> 不行的啊

抱歉看错了，看成需要多层级菜单了，同样的在分类管理那里拖拽就行

<img width="534" alt="image" src="https://github.com/user-attachments/assets/a59a9bd3-7ec4-4455-b9e5-fa06c7cffb35">
